### PR TITLE
Allow master key to override ACL restriction

### PR DESCRIPTION
**Pull request by ben181231**
connect #22

### CODE REVIEW

File: handler/record.go [hunk 1]
```diff
@@ -298,6 +298,7 @@ func (h *RecordSaveHandler) Handle(payload *router.Payload, response *router.Res
 		UserInfo:      payload.UserInfo,
 		RecordsToSave: p.Records,
 		Atomic:        p.Atomic,
+		WithMasterKey: payload.HasMasterAccess(),
```
Comment (cheungpat) [inline handler/record.go]: On one hand you have a the `router` package hide the fact that the user has master key and expose whether the user has master access, but later in the handler, the fact that the user has master access is converted to say the user has a master key.

If having master access does not mean the user have master key, this line is wrong.
If having master access means the user have master key, there is no need for such distinction, and `HasMasterAccess` should be called `HasMasterKey` instead

Comment (ben181231) [inline handler/record.go]: Got it.

Comment (ben181231) [inline handler/record.go]: I will rename it to `HasMasterKey`

**Comment by ben181231** [comment]
@cheungpat Updated

**Pull request merged**